packages/replay/src/eventBuffer.ts

```diff
@@ -73,14 +73,22 @@ class EventBufferArray implements EventBuffer {
 
   public finish(): Promise<string> {
     return new Promise<string>(resolve => {
-      // Make a copy of the events array reference and immediately clear the
-      // events member so that we do not lose new events while uploading
-      // attachment.
-      const eventsRet = this._events;
-      this._events = [];
-      resolve(JSON.stringify(eventsRet));
+      resolve(this._finish());
     });
   }
+
+  public finishImmediate(): string {
```

Comment:
After thinking about this, let's go with `finishSync` here maybe? That's maybe a bit clearer in what it does?
Also I we can save a few bytes by having `finishSync` be the actual implementation (instead of `_finish`) and using this in `finish()`.